clash-prelude/clash-prelude.cabal

```diff
@@ -335,7 +335,8 @@ Library
                       recursion-schemes         >= 5.1     && < 5.3,
                       QuickCheck                >= 2.7     && < 2.15,
                       reflection                >= 2       && < 2.2,
-                      singletons                >= 1.0     && < 3.0,
+                      singletons                >= 3.0     && < 4.0,
+                      singletons-base           >= 3.0     && < 4.0,
```

Comment:
Why do we need `singletons-base`?

Reply:
I don't think we do: the only things this PR exports from `Prelude.Singletons` are things re-exported from `Data.Singletons` in `singletons-3.0`

Reply:
... and we don't appear to anything which is only in `singletons-base` anywhere else, so we should be able to drop it